src/Core/Content/Media/Core/Application/RemoteThumbnailLoader.php

```diff
@@ -168,19 +174,33 @@ private function getBaseUrl(): string
         return \rtrim($this->filesystem->publicUrl(''), '/');
     }
 
-    private function getUrl(string $mediaUrl, string $mediaPath, string $width, string $height, ?\DateTimeInterface $mediaUpdatedAt): string
+    private function getUrl(MediaEntity|PartialEntity $mediaEntity, string $mediaUrl, string $width, string $height): ?string
     {
         return $this->extensions->publish(
             name: ResolveRemoteThumbnailUrlExtension::NAME,
             extension: new ResolveRemoteThumbnailUrlExtension(
                 $mediaUrl,
-                $mediaPath,
+                $mediaEntity->get('path'),
                 $width,
                 $height,
                 $this->pattern,
-                $mediaUpdatedAt
+                $mediaEntity->get('updatedAt') ?? $mediaEntity->get('createdAt'),
+                $mediaEntity,
             ),
-            function: function (string $mediaUrl, string $mediaPath, string $width, string $height, string $pattern, ?\DateTimeInterface $mediaUpdatedAt) {
+            function: function (
+                string $mediaUrl,
+                string $mediaPath,
+                string $width,
+                string $height,
+                string $pattern,
+                ?\DateTimeInterface $mediaUpdatedAt,
+                Entity $mediaEntity,
+            ): string {
+                if (!Feature::isActive('v6.8.0.0')) {
```

Comment:
```suggestion
                if (Feature::isActive('v6.8.0.0')) {
```

we should only do that if the feature flag for 6.8 is active, right? so this is the new way, from the next major version on